src/api/app/services/meta_controller_service/meta_validator.rb

```diff
@@ -0,0 +1,22 @@
+module MetaControllerService
+  class MetaValidator
```

Comment:
I don't think the class name reflects what this class is doing, it does not validate the Meta but the XML, so ``MetaXmlValidator`` might be a better name.